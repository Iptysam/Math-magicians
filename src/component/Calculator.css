* {
  padding: 0;
  margin: 0;
}

.main {
  background: #fff;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.min-bar {
  width: 92.5%;
  height: 100%;
  border: none;
  color: #fff;
  padding: 10px;
  text-align: right;
  background: grey;
}

.row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: 0.0625rem solid #222;
  padding: 0.625rem 1.25rem;
  color: black;
  text-align: center;
  text-decoration: none;
  font-size: 1.0625rem;
}

button:active {
  background-color: #ccc;
}

.color {
  background: orange;
}

.zero {
  grid-column: 1/ span 2;
}
