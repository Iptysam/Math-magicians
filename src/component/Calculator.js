import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    const num = e.target.textContent;
    const ans = calculate(state, num);
    setState(ans);
  };

  const { total, next, operation } = state;
  return (
    <div className="main">
      <div className="min-calculator">
        <div className="min-bar">
          {total}
          {operation}
          {next}
        </div>
        <div className="first row">
          <button type="button" onClick={handleClick}>AC</button>
          <button type="button" onClick={handleClick}>+/-</button>
          <button type="button" onClick={handleClick}>%</button>
          <button type="button" onClick={handleClick} className="color">รท</button>
        </div>
        <div className="2nd row">
          <button type="button" onClick={handleClick}>7</button>
          <button type="button" onClick={handleClick}>8</button>
          <button type="button" onClick={handleClick}>9</button>
          <button type="button" onClick={handleClick} className="color">ร</button>
        </div>
        <div className="3rd row">
          <button type="button" onClick={handleClick}>4</button>
          <button type="button" onClick={handleClick}>5</button>
          <button type="button" onClick={handleClick}>6</button>
          <button type="button" onClick={handleClick} className="color">-</button>
        </div>
        <div className="4th row">
          <button type="button" onClick={handleClick}>1</button>
          <button type="button" onClick={handleClick}>2</button>
          <button type="button" onClick={handleClick}>3</button>
          <button type="button" onClick={handleClick} className="color">+</button>
        </div>
        <div className="5th row">
          <button type="button" onClick={handleClick} className="zero">0</button>
          <button type="button" onClick={handleClick}>.</button>
          <button type="button" onClick={handleClick} className="color">=</button>
        </div>
      </div>
    </div>

  );
}

export default Calculator;
