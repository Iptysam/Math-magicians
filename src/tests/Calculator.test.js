import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../component/Calculator';

it('It should render the display of button in the screen', () => {
  const screen = renderer
    .create(<Calculator />)
    .toJSON();
  expect(screen).toMatchSnapshot();
});
