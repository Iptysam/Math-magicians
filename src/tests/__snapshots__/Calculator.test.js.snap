// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the display of button in the screen 1`] = `
<div
  className="main-page"
>
  <h1>
    Math Magicians
  </h1>
  <h2>
    Lets Do Some Math
  </h2>
  <div
    className="main"
  >
    <div
      className="min-calculator"
    >
      <div
        className="min-bar"
      >
        0
      </div>
      <div
        className="first row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="color"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="2nd row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="color"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="3rd row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="color"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="4th row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="color"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="5th row"
      >
        <button
          className="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="color"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`It should render the value on the screen`] = `

<div
 className="main-page"
 >
      <h1
      >
      Math Magicians
      </h1>
      <h2
      >
      Lets Do Some Math
      </h2>
      <div
       className="main"
       >
        <div
         className="min-calculator"
         >
          <div 
          className="min-bar"
          >
            {total}
            {operation}
            {next}
          </div>
          <div
           className="first row"
           >
            <button
             type="button"
             onClick={handleClick}
             >
             AC
             </button>
            <button
            type="button"
            onClick={handleClick}
            >
            +/-
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            %
            </button>
            <button
            type="button"
            onClick={handleClick}
            className="color"
            >
            รท
            </button>
          </div>
          <div
          className="2nd row"
          >
            <button
            type="button"
            onClick={handleClick}
            >
            7
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            8
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            9
            </button>
            <button
            type="button"
            onClick={handleClick}
            className="color"
            >
            x
            </button>
          </div>
          <div
          className="3rd row"
          >
               <button
            type="button"
            onClick={handleClick}
            >
            4
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            5
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            6
            </button>
            <button
            type="button"
            onClick={handleClick}
            className="color"
            >
            -
            </button>
          </div>
          <div
          className="4th row"
          >
               <button
            type="button"
            onClick={handleClick}
            >
            1
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            2
            </button>
            <button
            type="button"
            onClick={handleClick}
            >
            3
            </button>
            <button
            type="button"
            onClick={handleClick}
            className="color"
            >
            +
            </button>
          </div>
          <div
           className="5th row"
           >
            <button
             type="button"
              onClick={handleClick}
               className="zero"
               >
               0
               </button>
            <button
             type="button"
              onClick={handleClick}
              >
              .
              </button>
            <button
             type="button"
              onClick={handleClick}
               className="color"
               >
               =
               </button>
          </div>
        </div>
      </div>
</div>`;
